src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java

```diff
@@ -0,0 +1,708 @@
+
+            PendingImportObject<BrAPILocation> locationPIO = createLocationPIO(importRow);
+            this.locationByName.put(importRow.getEnvLocation(), locationPIO);
+
+            PendingImportObject<BrAPIStudy> studyPIO = createStudyPIO(program, commit, expSeqValue, importRow);
+            this.studyByNameNoScope.put(importRow.getEnv(), studyPIO);
+
+            PendingImportObject<BrAPIObservationUnit> obsUnitPIO = createObsUnitPIO(program, commit, obsUnitNextVal, importRow);
+            String key = createObservationUnitKey(importRow);
+            this.observationUnitByNameNoScope.put(key, obsUnitPIO);
+        }
+    }
+
+    private String createObservationUnitKey(ExperimentObservation importRow) {
+        String key = importRow.getEnv() + importRow.getExpUnitId();
+        return key;
+    }
+
+    private ValidationErrors validateFields(List<BrAPIImport> importRows, ValidationErrors validationErrors) {
+        HashSet<String> uniqueStudyAndObsUnit = new HashSet<>();
+        for (int i = 0; i < importRows.size(); i++) {
+            ExperimentObservation importRow = (ExperimentObservation) importRows.get(i);
+            validateConditionallyRequired(validationErrors, i, importRow);
+            validateUniqueObsUnits(validationErrors, uniqueStudyAndObsUnit, i, importRow);
+        }
+        return validationErrors;
+    }
+
+    /**
+     * Validate that the the observation unit is unique within a study.
+     * <br>
+     * SIDE EFFECTS:  validationErrors and uniqueStudyAndObsUnit can be modified.
+     * @param validationErrors can be modified as a side effect.
+     * @param uniqueStudyAndObsUnit can be modified as a side effect.
+     * @param i counter that is always two less the file row being validated
+     * @param importRow the data row being validated
+     */
+    private void validateUniqueObsUnits(
+            ValidationErrors validationErrors,
+            HashSet<String> uniqueStudyAndObsUnit,
+            int i,
+            ExperimentObservation importRow) {
+        String envIdPlusStudyId = createObservationUnitKey( importRow );
+        if( uniqueStudyAndObsUnit.contains( envIdPlusStudyId )){
+            String errorMessage = String.format("The ID (%s) is not unique within the environment(%s)", importRow.getExpUnitId(), importRow.getEnv());
+            this.addRowError("Exp Unit ID", errorMessage, validationErrors, i);
+        }
+        else{
+            uniqueStudyAndObsUnit.add( envIdPlusStudyId );
+        }
+    }
+
+    private void validateConditionallyRequired(ValidationErrors validationErrors, int i, ExperimentObservation importRow) {
+        String experimentTitle = importRow.getExpTitle();
+        String obsUnitID = importRow.getObsUnitID();
+        if( StringUtils.isBlank( obsUnitID )){
+            validateRequiredCell(
+                    experimentTitle,
+                    "Exp Title",
+                    "Field is blank", validationErrors, i
+                    );
+        }
```

Comment:
I believe this should be checking `experimentTitle` not `obsUnitId`.

Reply:
I think this logic is correct.  The method call 
`validateRequiredCell(
                    experimentTitle,
                    "Exp Title",
                    "Field is blank", validationErrors, i
                    );`
will throw an error if _experimentTitle_ is blank

Reply:
Ok, I think I'm understanding. Based on this line:

```
if( StringUtils.isBlank( obsUnitID )){
```

In order to throw the ExpTitle is blank error, the obsUnitId needs to be blank as well? If the obsUnitId is provided, that means we are important observations, not an experiment?